src/qt/bitcoin.cpp

```diff
@@ -676,7 +682,7 @@ int main(int argc, char *argv[])
     app.createOptionsModel(IsArgSet("-resetguisettings"));
 
     // Subscribe to global signals from core
-    uiInterface.InitMessage.connect(InitMessage);
+    FIXME_IMPLEMENT_IPC_VALUE(uiInterface).InitMessage.connect(InitMessage);
```

Comment:
> How would you handle uiInterface in this model, e.g. signals from the server to the client?

This line needs to change to `ipcNode.handleInitMessage(InitMessage)`. If you look at the `ipc::Node::handleInitMessage` implementation, it takes a std::function argument, creates a capnp `InitMessageCallback::Server` object that can invoke it, and sends a reference to that object over the IPC channel to `bitcoind`. `bitcoind` then calls `uiInterface.InitMessage.connect` with a handler that sends messages with the `InitMessageCallback` object.